c/driver/postgresql/postgres_copy_reader_test.cc

```diff
@@ -136,21 +137,19 @@ TEST(PostgresCopyUtilsTest, PostgresCopyWriteBoolean) {
   PostgresCopyStreamWriteTester tester;
   ASSERT_EQ(tester.Init(&schema.value, &array.value), NANOARROW_OK);
 
-  struct ArrowBuffer buffer;
-  ArrowBufferInit(&buffer);
-  ArrowBufferReserve(&buffer, sizeof(kTestPgCopyBoolean));
-  uint8_t* cursor = buffer.data;
+  ArrowBufferInit(&buffer.value);
+  ArrowBufferReserve(&buffer.value, sizeof(kTestPgCopyBoolean));
+  uint8_t* cursor = buffer->data;
```

Comment:
Still not sure of the API here; right now things are definitely broken if we do not provide a suitably sized buffer into the writer, and the writer in its current implementation advances the `data` pointer member of the provided buffer. 

The test at least resets the data member to its original position so things are properly free'd, but not quite sure how to handle the "needs resize" case (for another PR)

Reply:
Ok thanks for the ideas. I think having the writer manage internally might be best, especially in a future where we implement chunked writes. Let me give that a pass here